config/schemas/config.yaml

```diff
@@ -7,13 +7,245 @@ description: |
   cluster and the workload cluster, but keep in mind that each configuration
   file will contain different settings.
 $defs:
-  $comment: |
-    Location for common types of things for reuse with a reference like
+  kubernetesAffinity:
+    $comment: Real k8s resources might have existing schemas that could be reused as-is
+    properties:
+      nodeAffinity:
+        type: object
+      podAffinity:
+        type: object
+      podAntiAffinity:
+        type: object
+    additionalProperties: false
+    examples:
+      - nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+              - matchExpressions:
+                  - key: node-role.kubernetes.io/control-plane
+                    operator: Exists
+    title: Kubernetes Affinity
+    type: object
```

Comment:
```diff
diff --git a/config/schemas/config.yaml b/config/schemas/config.yaml
index b10d2a9f..9c3fe3f9 100644
--- a/config/schemas/config.yaml
+++ b/config/schemas/config.yaml
@@ -8,7 +8,6 @@ description: |
   file will contain different settings.
 $defs:
   kubernetesAffinity:
-    $comment: Real k8s resources might have existing schemas that could be reused as-is
     properties:
       nodeAffinity:
         type: object
@@ -19,11 +18,7 @@ $defs:
     additionalProperties: false
     examples:
       - nodeAffinity:
-          requiredDuringSchedulingIgnoredDuringExecution:
-            nodeSelectorTerms:
-              - matchExpressions:
-                  - key: node-role.kubernetes.io/control-plane
-                    operator: Exists
+          $ref: https://github.com/kubernetes/kubernetes/raw/master/api/openapi-spec/v3/api__v1_openapi.json#/components/schemas/io.k8s.api.core.v1.NodeAffinity
     title: Kubernetes Affinity
     type: object
   component:
```
🤔

Reply:
Heh, accidentally put the `$ref` under `examples`. Now fixed.

I'm a bit uneasy about referencing remote files tho, and not sure if licensing allows copying subsets.

Reply:
Should be fine to copy the subset with proper attribution given that the upstream spec is Apache-2.0, so if you add a comment in the spec where you include it and a notice in the readme then it should be OK.

Reply:
Those affinity settings contain further `$ref`erences which point to things with even more references, turning this into a recursive chase that may need further though or some scripting.

Reply:
Hooray for loops!

```
     [ck8s] Failed schema validation:
     wc-config.yaml: hnc.manager.nodeSelector: nodeSelectorTerms is required
     wc-config.yaml: velero.nodeSelector: nodeSelectorTerms is required
```

What to do with these? This could be due to importing definitions from latest k8s git. I haven't investigated yet.

Reply:
Oh no, `NodeSelector` and `nodeSelector` are different things.

Reply:
I've made it so that `.$defs.kubernetesThing` are kubernetes properties where we wrote the schema, while those imported from the kubernetes API spec files have names like `.$defs["io.k8s.api.core.v1.Affinity"]`.